### ISSUE TITLE
Changing subscription's plan ignores quantity

### ISSUE THREAD

**Issue by alanhamlett** [stripe/stripe-python#388]
## Description

When a customer has an existing subscription with quantity 2 and plan `plan1`, updating the subscription's plan always uses quantity 1.

## Example

```python
subscription = stripe.Subscription.retrieve({SUBSCRIPTION_ID})
assert subscription.plan == 'plan1'
assert subscription.quantity == 2
subscription.plan = 'plan2'
subscription.quantity = 2
subscription.save()

subscription = stripe.Subscription.retrieve({SUBSCRIPTION_ID})
assert subscription.quantity == 2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

Stripe logging shows only `plan_id` is sent to the api when calling `subscription.save()`.

**Comment by remi-stripe** [comment]
@alanhamlett Thanks a lot for the report! I think that the issue here is that `quantity` does not change so it's not sent to the API. On the other hand, server-side the API resets `quantity` to 1 when you change the plan id.

We'll likely need to special-case this in the library. Assigning to @ob-stripe who was looking into it!

**Comment by ob-stripe** [comment]
This issue is caused by the fact that the Python library will only register an attribute as an unsaved value when it is updated with a different value:

```python
>>> s = stripe.Subscription.retrieve('sub_...')
>>> s.quantity
1
>>> s.quantity = 1
>>> s._unsaved_values
set([])
>>> s.quantity = 2
set(['quantity'])
```

This behavior is different from most of our other libraries, where attributes are registered as unsaved on assignment, regardless of the new value. E.g. in Ruby:

```ruby
[1] pry(main)> s = Stripe::Subscription.retrieve('sub_BqStq3lplpOQKs')
[2] pry(main)> s.quantity
=> 1
[3] pry(main)> s.quantity = 1
[4] pry(main)> s.instance_variable_get(:@unsaved_values)
=> #<Set: {:quantity}>
```

I'll push a fix, but in the meantime a simple workaround is to set a different value, then immediately set it back. E.g. in your case:
```python
subscription.plan = 'plan2'
subscription.quantity = 42
subscription.quantity = 2
subscription.save()
```

**Comment by brandur-stripe** [comment]
Pushed a fix with @ob-stripe's patch in 1.77.1. Thanks @ob-stripe / @remi-stripe!

**Comment by alanhamlett** [comment]
Thanks!
